### PR TITLE
LPS-70863 Allow to dynamically add or remove form validation rules

### PR DESCRIPTION
**Pull request by jbalsas**

### CODE REVIEW

**Comment by brunobasto** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by brunobasto** [comment]
Hey @ambrinchaudhary, this looks good! I think these lines of code could also benefit from this new API: https://github.com/liferay/liferay-portal/blob/master/modules/apps/forms-and-workflow/dynamic-data-mapping/dynamic-data-mapping-web/src/main/resources/META-INF/resources/js/ddm_form.js#L2982

Should I send it back to you or forward it to bchan?

File: modules/apps/foundation/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/form.js [hunk 1]
```diff
@@ -123,6 +123,19 @@ AUI.add(
 						}
 					},
 
+					_findRuleIndex: function(fieldRules, fieldName, validatorName) {
+						var instance = this;
+
+						var index = fieldRules.findIndex(
```
Comment (brunobasto) [inline modules/apps/foundation/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/form.js]: @ambrinchaudhary we could probably immediately return here.

Comment (ambrinchaudhary) [inline modules/apps/foundation/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/form.js]: 👍

**Comment by ambrinchaudhary** [comment]
Hey @brunobasto, yes, in the lines that you mention we could now use these new methods. 
Please, change the `_findRuleIndex` method as you suggested and forward it to Brian. Thanks!

**Comment by brunobasto** [comment]
Pull request submitted to https://github.com/brianchandotcom/liferay-portal/pull/47274. [See changes here.](https://github.com/brunobasto/liferay-portal/compare/f0126d1~1...f0126d1) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Pull request closed without merging**